### PR TITLE
Create a more precise closing of the location log on radius accounting stop

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Create a more precise closing of the location log on radius accounting stop
## NEWS file entries

Enhancements
++++++++++++
- Allow the locationlog to be closed on an accounting stop
## Delete branch after merge

NO

### CODE REVIEW

File: conf/pf.conf.defaults [hunk 1]
```diff
@@ -1092,6 +1092,11 @@ ldap_attributes=uid,cn,department,displayName,distinguishedName,givenName,member
 #
 # Amount of pfdns processes to start
 pfdns_processes=4
+#
+# advanced.locationlog_close_on_accounting_stop
+#
+# Close the locationlog for a node on accounting stop
+locationlog_close_on_accounting_stop=disabled
```
Comment (fdurand) [inline conf/pf.conf.defaults]: enabled by default

File: lib/pf/radius.pm [hunk 2]
```diff
@@ -308,7 +308,9 @@ sub accounting {
             #handle radius floating devices
             $self->_handleAccountingFloatingDevices($switch, $mac, $port);
             #Close old locationlog
-            locationlog_update_end_mac($mac);
+            if (enabled($Config{advanced}{'locationlog_close_on_accounting_stop'})) {
```
Comment (fdurand) [inline lib/pf/radius.pm]: isenabled

File: lib/pf/locationlog.pm [hunk 3]
```diff
@@ -414,6 +420,28 @@ sub locationlog_insert_closed {
     return (1);
 }
 
+=item locationlog_update_end_mac_switch_port
+
+End an open locationlog based off mac, switch and port
```
Comment (fdurand) [inline lib/pf/locationlog.pm:425]: of

Comment (rouzier) [inline lib/pf/locationlog.pm:425]: Not a typo

**Comment by fdurand** [comment]
Rebase please

File: lib/pf/locationlog.pm [hunk 4]
```diff
@@ -414,6 +420,28 @@ sub locationlog_insert_closed {
...
+sub locationlog_update_end_mac_switch_port {
+    my ( $mac, $switch, $port ) = @_;
+    my $logger = get_logger();
+    if (!defined $mac || !defined $switch || !defined $port ) {
+        $logger->error("Invalid parameters provided mac=" . (defined $mac ? $mac : "undefined")
+              . " switch=" . (defined $switch ? $switch : "undefined")
+              . " port=" . (defined $port ? $port : "undefined"));
+        return (0);
+    }
+    $logger->info("locationlog_update_end_mac_switch_port called with mac=$mac switch=$switch port=$port");
```
Comment (fdurand) [inline lib/pf/locationlog.pm]: Change it to debug

**Comment by jrouzierinverse** [comment]
Done please review

**Pull request merged**